mixer/test/client/pilotplugin/pilotplugin_test.go

```diff
@@ -0,0 +1,377 @@
+    connect_timeout: 5s
+    type: STATIC
+    hosts:
+    - socket_address:
+        address: 127.0.0.1
+        port_value: {{.Ports.ServerProxyPort}}
+  - name: "inbound|||backend"
+    connect_timeout: 5s
+    type: STATIC
+    hosts:
+    - socket_address:
+        address: 127.0.0.1
+        port_value: {{.Ports.BackendPort}}
+  - name: "outbound|9091||mixer_server"
+    http2_protocol_options: {}
+    connect_timeout: 5s
+    type: STATIC
+    hosts:
+    - socket_address:
+        address: 127.0.0.1
+        port_value: {{.Ports.MixerPort}}
+`
+
+const checkAttributesOkGet = `
+{
+  "connection.mtls": false,
+  "context.protocol": "http",
+  "context.reporter.kind": "inbound",
+  "context.reporter.uid": "kubernetes://pod1.ns2",
+  "destination.ip": "[0 0 0 0 0 0 0 0 0 0 255 255 127 0 0 1]",
+  "destination.port": "*",
+  "destination.namespace": "ns2",
+  "destination.uid": "kubernetes://pod1.ns2",
+  "destination.service": "svc.ns3",
+  "destination.service.host": "svc.ns3",
+  "destination.service.name": "svc",
+  "destination.service.namespace": "ns3",
+  "destination.service.uid": "svcuid",
+  "source.uid": "kubernetes://pod2.ns2",
+  "request.headers": {
+     ":method": "GET",
+     ":path": "/echo",
+     ":authority": "*",
+     "x-forwarded-proto": "http",
+     "x-request-id": "*"
+  },
+  "request.host": "*",
+  "request.path": "/echo",
+  "request.time": "*",
+  "request.useragent": "Go-http-client/1.1",
+  "request.method": "GET",
+  "request.scheme": "http"
+}
+`
+const reportAttributesOkOutbound = `
+{
+  "connection.mtls": false,
+  "context.protocol": "http",
+  "context.reporter.kind": "outbound",
+  "context.reporter.uid": "kubernetes://pod2.ns2",
+  "destination.ip": "[127 0 0 1]",
+  "destination.port": "*",
+  "destination.service": "svc.ns3",
+  "destination.service.host": "svc.ns3",
+  "destination.service.name": "svc",
+  "destination.service.namespace": "ns3",
+  "destination.service.uid": "svcuid",
+  "source.uid": "kubernetes://pod2.ns2",
+  "source.namespace": "ns2",
+  "request.headers": {
+     ":method": "GET",
+     ":path": "/echo",
+     ":authority": "*",
+     "x-forwarded-proto": "http",
+     "x-istio-attributes": "-",
+     "x-request-id": "*"
+  },
+  "request.host": "*",
+  "request.path": "/echo",
+  "request.time": "*",
+  "request.useragent": "Go-http-client/1.1",
+  "request.method": "GET",
+  "request.scheme": "http",
+  "request.size": 0,
+  "request.total_size": "*",
+  "response.time": "*",
+  "response.size": 0,
+  "response.duration": "*",
+  "response.code": 200,
+  "response.headers": {
+     "date": "*",
+     "content-length": "0",
+     ":status": "200",
+     "server": "envoy"
+  },
+  "response.total_size": "*"
+}`
+
+const reportAttributesOkInbound = `
+{
+  "connection.mtls": false,
+  "context.protocol": "http",
+  "context.reporter.kind": "inbound",
+  "context.reporter.uid": "kubernetes://pod1.ns2",
+  "destination.ip": "[0 0 0 0 0 0 0 0 0 0 255 255 127 0 0 1]",
```

Comment:
@douglas-reid golang started using 16 byte IPv6 for IPv4 byte arrays. Not sure if it matters, but it's ugly.